DBCON-299: details on different classloader setup models (#1701)
mulesoft/docs-connectors: db/1.10/modules/ROOT/pages/database-connector-driver-configuration.adoc

= Configure Driver Visibility using Mule Maven Plugin

> Anypoint Studio helps you configure database driver dependencies without problems, though there are some scenarios where you must use the Mule Maven Plugin to set up properly the driver classes visibility because of class loading isolation mechanisms.

Reviewer: ```suggestion
Although Anypoint Studio enables you to configure database driver dependencies without problems in most cases, there are some scenarios where you must use the Mule Maven Plugin to properly set up the driver classes' visibility because of class loading isolation mechanisms.
```

== Configure Driver Visibility Across the Application

> To configure additional database driver dependencies and make the driver visible across components within your Mule application, refer to the xref:mule-runtime::mmp-concept.adoc#configure-shared-libraries[configure shared libraries] section in the Mule Maven Plugin documentation.

Reviewer: ```suggestion
To configure additional database driver dependencies and make the driver visible across components within your Mule application, refer to the xref:mule-runtime::mmp-concept.adoc#configure-shared-libraries[Configure Shared Libraries] section in the Mule Maven Plugin documentation.
```

== Configure Driver Visibility Only for the Connector

> When there is a risk of dependencies collision or there is a necessity to have Database Connector and the driver dependency in the same class loader, which makes the driver visible only for the connector, refer to the xref:mule-runtime::mmp-concept.adoc#configure-plugin-dependencies[configure additional plugin dependencies] section in the Mule Maven Plugin documentation.

Reviewer: ```suggestion
When there is a risk of dependencies colliding and it's necessary to have the Database Connector and the driver's dependencies in the same class loader, to make the driver visible only for the connector, refer to the xref:mule-runtime::mmp-concept.adoc#configure-plugin-dependencies[Configure Additional Plugin Dependencies] section in the Mule Maven Plugin documentation.
```

[NOTE]

> Be aware of which component must establish the connection with the database. The driver must be in the scope of that component's class loader. For example, a Mule application that uses datasource reference connection and targets a spring bean datasource definition. It is the spring's class loader that requires the database library, not the database. Configure additional application dependencies using _shared libraries_.

Reviewer: ```suggestion
Be aware of which component must establish the connection with the database, as the driver must be in the scope of that component's class loader. For example, if a Mule application uses a datasource reference connection and targets a Spring bean datasource definition, it is the Spring's class loader that requires the database library, not the database. Configure additional application dependencies using _shared libraries_.
```
